File: src/main/java/com/robohorse/robopojogenerator/generator/consts/annotations/PojoAnnotations.java
package com.robohorse.robopojogenerator.generator.consts.annotations;

/**
 * Created by vadim on 02.10.16.
 */
public interface PojoAnnotations {
    String INFO_ANNOTATION = "";

    interface GSON {
        String CLASS_ANNOTATION = INFO_ANNOTATION;
        String ANNOTATION = "@SerializedName(\"%1$s\")";
    }

    interface LOGAN_SQUARE {
        String CLASS_ANNOTATION = INFO_ANNOTATION + "\n@JsonObject";
        String ANNOTATION = "@SerializedName(\"%1$s\")\n\t@JsonField(name =\"%1$s\")";
    }

    interface JACKSON {
        String CLASS_ANNOTATION = INFO_ANNOTATION;
        String ANNOTATION = "@JsonProperty(\"%1$s\")";
    }

    interface AUTO_VALUE_GSON {
        String CLASS_ANNOTATION = INFO_ANNOTATION + "\n@AutoValue";
        String ANNOTATION = "@SerializedName(\"%1$s\")";
    }

    interface FAST_JSON {
        String CLASS_ANNOTATION = INFO_ANNOTATION;
        String ANNOTATION = "@JSONField(name=\"%1$s\")";
    }

    interface MOSHI {
        String CLASS_ANNOTATION = INFO_ANNOTATION;
        String ANNOTATION = "@Json(name = \"%1$s\")";
    }
}

File: src/main/kotlin/com/robohorse/robopojogenerator/generator/consts/annotations/PojoAnnotations.kt
package com.robohorse.robopojogenerator.generator.consts.annotations

sealed class PojoAnnotations(
        val classAnnotation: String = EMPTY_ANNOTATION,
        val annotation: String = EMPTY_ANNOTATION
) {
    object GSON : PojoAnnotations(
            annotation = "@SerializedName(\"%1\$s\")"
    )

    object LOGAN_SQUARE : PojoAnnotations(
            classAnnotation = "@JsonObject",
            annotation = "@JsonField(name =\"%1\$s\")"
    )

    object JACKSON : PojoAnnotations(
            annotation = "@JsonProperty(\"%1\$s\")"
    )

    object AUTO_VALUE_GSON : PojoAnnotations(
            classAnnotation = "@AutoValue",
            annotation = "@SerializedName(\"%1\$s\")"
    )

    object FAST_JSON : PojoAnnotations(
            annotation = "@JSONField(name=\"%1\$s\")"
    )

    object MOSHI : PojoAnnotations(
            annotation = "@Json(name = \"%1\$s\")"
    )

    object LOMBOK : PojoAnnotations()
}

private const val EMPTY_ANNOTATION = ""
